### PR TITLE
refactor: migrate remaining console.log in lint.ts to scoped logger

### LINKED ISSUE #35: Migrate remaining console.log calls in lint.ts to scoped logger
## Problem

`src/lint.ts` has ~80 `console.log` calls mixed with `getScopedLogger()` calls. The rest of the codebase has been migrated to scoped logging.

Some functions use `getScopedLogger` (e.g., `LintDuplicateCategoryIds`, `LintManualIdHygiene`, `LintIconsSheet`), while others still use raw `console.log`.

## Impact

- Inconsistent logging makes debugging harder
- No log levels for most lint operations
- Cannot filter lint logs by scope

## Fix

Replace all `console.log` calls with appropriate scoped logger calls:

```typescript
const logger = getScopedLogger("LintCategories");
logger.info("Categories sheet linting completed");
```

## Effort

~2 hours (mechanical find-and-replace with scope naming)

### PR DESCRIPTION
**Pull request by luandro**
## Problem

`src/lint.ts` had ~50 `console.log`/`console.warn`/`console.error`/`console.time` calls mixed with `getScopedLogger()` calls. The rest of the codebase has been fully migrated to scoped logging.

## Changes

Replaced all non-test `console.*` calls with appropriate `getScopedLogger()` calls across 17 functions:

- **LintWhitespace** — cleanWhitespaceOnlyCells
- **LintDuplicates** — checkForDuplicates
- **LintUnreferencedDetails** — checkUnreferencedDetails
- **LintDetails** — validateUniversalFlag, lintDetailsSheet
- **LintTranslationSlugs** — checkDuplicateTranslationSlugs
- **LintTranslationHeaders** — validateTranslationHeaders
- **LintSheet** — lintSheet
- **LintCategoryIcons** — validateCategoryIcons
- **LintCategories** — lintCategoriesSheet
- **LintTranslationSourceOverwrites** — checkTranslationSourceOverwrites
- **LintTranslations** — lintTranslationSheets
- **LintTranslationConsistency** — validateTranslationSheetConsistency, validateSheetConsistency
- **LintDialogHtml** — validateDialogHtml
- **LintFixTranslationSheets** — fixTranslationMismatches
- **LintAllSheets** — lintAllSheets

### Log level mapping
- Starting/completing operations → `logger.info`
- Warnings/edge cases → `logger.warn`
- Errors → `logger.error`
- Debug details → `logger.debug`
- `console.time`/`console.timeEnd` → `logger.debug` (GAS does not support console.time)

### Not modified
- Test functions (`testHtmlValidation`, `testDialogHtmlGeneration`, `runAllHtmlValidationTests`) — intentional test output preserved

No behavior changes. Pure logging migration.

Closes #35

<h3>Greptile Summary</h3>

This PR migrates all remaining `console.log`/`console.warn`/`console.error`/`console.time` calls in `src/lint.ts` to the project's `getScopedLogger()` API, completing the codebase-wide structured logging migration. Log level mapping follows the documented convention (info for lifecycle events, warn for edge cases, error for failures, debug for detail/former timing markers).

- **17 functions updated** across the lint pipeline, each receiving a scoped logger (mostly hoisted to the function entry point) so log output carries a consistent `[ScopeName]` prefix.
- **`console.time`/`console.timeEnd` pairs removed** in `lintSheet` (including the `finally` block) and replaced with `logger.debug` start markers; GAS does not support `console.time`, making this a correctness improvement on that runtime.
- **Test helper functions left untouched**, preserving intentional `console.*` output in `testHtmlValidation`, `testDialogHtmlGeneration`, and `runAllHtmlValidationTests`.

<h3>Confidence Score: 4/5</h3>

Safe to merge; this is a pure logging substitution with no changes to lint logic, sheet mutations, or return values.

All 17 functions are cleanly migrated and logger placement is correct at function entry in the vast majority of cases. An open thread from a prior review notes that the logger in checkDuplicateTranslationSlugs is created inside the inner column loop, making it technically out of scope for the catch block — that remains unresolved in this version. No other logic is touched.

src/lint.ts — specifically the checkDuplicateTranslationSlugs function where the logger scope concern from the prior review is still open.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/lint.ts | Pure logging refactor replacing ~50 console.* calls with getScopedLogger(); logger placement is correct at function entry in most cases, with two pre-existing scope-placement concerns already flagged in earlier review threads. |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[lint.ts function called] --> B[getScopedLogger at function entry]
    B --> C{Operation type}
    C -->|lifecycle start/end| D[logger.info]
    C -->|edge case or empty sheet| E[logger.warn]
    C -->|exception caught| F[logger.error]
    C -->|diagnostic detail or former console.time| G[logger.debug]
    D --> H[AppAppLogger.scope prepends ScopeName]
    E --> H
    F --> H
    G --> H
    H --> I[console.log / console.warn / console.error via AppAppLogger.output]
```

<details open><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `src/lint.ts`, line 1370-1379 ([link](https://github.com/digidem/comapeo-category-spreadsheet-plugin/blob/83f00673ea10e52dce1a00154b839ca48433b917/src/lint.ts#L1370-L1379)) 

   <a href="#">P2</a> The logger is created separately in three different scopes inside `checkUnreferencedDetails` — once inside the early-exit `if`, once inside the `for` loop on every unreferenced entry, and once in the `catch` block. The per-iteration instantiation in the loop is the main concern: when `AppLogger` is live, `scope()` allocates a new logger object on each call. Hoist a single `getScopedLogger("LintUnreferencedDetails")` call to the top of the function.

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: src/lint.ts
   Line: 1370-1379

   Comment:
   The logger is created separately in three different scopes inside `checkUnreferencedDetails` — once inside the early-exit `if`, once inside the `for` loop on every unreferenced entry, and once in the `catch` block. The per-iteration instantiation in the loop is the main concern: when `AppLogger` is live, `scope()` allocates a new logger object on each call. Hoist a single `getScopedLogger("LintUnreferencedDetails")` call to the top of the function.

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

   Note: If this suggestion doesn't match your team's coding style, reply to this and let me know. I'll remember it for next time!
</details>

<sub>Reviews (2): Last reviewed commit: ["refactor: hoist getScopedLogger calls ou..."](https://github.com/digidem/comapeo-category-spreadsheet-plugin/commit/8e832e063eed43937df86a7074232b89d888927f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=35355276)</sub>

### CODE REVIEW

File: src/lint.ts [hunk 1]
```diff
@@ -1462,7 +1468,8 @@ function checkDuplicateTranslationSlugs(): void {
         // Highlight cells with duplicate slugs
         for (const [slug, rows] of slugCounts.entries()) {
           if (rows.length > 1) {
-            console.log(
+            const logger = getScopedLogger("LintTranslationSlugs");
+            logger.debug(
```
[the remark this answers is not in the document]
Comment (luandro) [reply src/lint.ts:1469]: Fixed in 8e832e0. Hoisted getScopedLogger out of loops in both checkDuplicateTranslationSlugs and checkUnreferencedDetails. Removed 4 redundant instantiations.

**Comment by luandro** [comment]
Superseded by PR #46 which now includes both the lint split AND the scoped logger migration. No need for a separate PR.
